native/src/main/java/io/ballerina/stdlib/io/nativeimpl/RecordChannelUtils.java

```diff
@@ -173,7 +174,7 @@ public static Object getAllRecords(BObject channel, int skipHeaders, BTypedesc t
                 Object[] out = outList.toArray();
                 return ValueCreator.createArrayValue(out, TypeCreator.createArrayType(describingType));
             }
-        } catch (BallerinaIOException e) {
+        } catch (BallerinaIOException | BError e) {
             return IOUtils.createError(e);
```

Comment:
Is there need to wrap the BError in another BError?

Reply:
For `BallerinaIOException` standard previously used procedure was to create a `BError` [example](https://github.com/ballerina-platform/module-ballerina-io/blob/8f56eb88e26c1679f7b1a2defe90bbf87c86da5e/native/src/main/java/io/ballerina/stdlib/io/nativeimpl/CharacterChannelUtils.java#L94). As we try to handle `BError` in addition to `BallerinaIOException`, [this](https://github.com/ballerina-platform/module-ballerina-io/blob/8f56eb88e26c1679f7b1a2defe90bbf87c86da5e/native/src/main/java/io/ballerina/stdlib/io/utils/IOUtils.java#L71) can be used to create a new Error using the `message` without wrapping a BError in a new BError.

Reply:
Yeah but it is an unnecessary logic,
```suggestion
        } catch (BallerinaIOException e) {
            return IOUtils.createError(e);
        } catch (BError e) {
            return e;
        }
```
This should be the correct fix. The issue is we are using throw statement in the first place. Look into if we can return instead of using throw.

Reply:
Yes, This is more simpler. Updated the function to return errors, rather than `throw` in [bfb41dd](https://github.com/ballerina-platform/module-ballerina-io/pull/669/commits/bfb41ddd50ad895a8585cfd9184655a0e4e11b9f)